### PR TITLE
Issue/3785 - Add alignment controls to heading block toolbar

### PR DESCRIPTION
**Pull request by amdrew**
## Description

Adds the alignment controls to the toolbar, as discussed in #3785

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.

### CODE REVIEW

File: blocks/library/heading/index.js [hunk 1]
```diff
@@ -117,7 +126,9 @@ export const settings = {
 							subscript: level,
 						} ) )
 					}
-				/>
+				>
+				{ alignmentToolbar }
```
Comment (aduth) [inline blocks/library/heading/index.js]: Can we tab this once more? (Ideally we should have an ESLint rule for this)

**Review by aduth — COMMENTED** [review]
Code-wise looks good.

>It seems we should also remove the alignment controls from the block inspector? If so, I'll update the PR.

I'd like some feedback here. Seems to me at least we'd want to drop the duplicated controls from the sidebar inspector. cc @jasmussen

File: blocks/library/heading/index.js (hunk 1 above)
Comment (amdrew) [reply blocks/library/heading/index.js]: Fixed

**Comment by jasmussen** [comment]
Thanks so much for working on this! Here's a screenshot:

screen shot 2018-04-03 at 08 36 00

This looks good, but it doesn't account for responsive or mobile breakpoints:

screen shot 2018-04-03 at 08 36 06

screen shot 2018-04-03 at 08 37 13

One suggestion detailed in https://github.com/WordPress/gutenberg/issues/3785#issuecomment-351412573 is to remove the 3 quick shortcuts to set h2 h3 and h4, and instead move those to the switcher interface. To here:

screen shot 2018-04-03 at 08 37 28

☝️ Imagine headings 1-6 all in that list. 

I realize that's a lot more work than just adding the alignments toolbar, and if this proves too much work for you then that's totally fine, we'll see if one of the core devs has time to work on it. 

I also agree with Andrew, we should remove both the heading level and alignments from the inspector:

screen shot 2018-04-03 at 08 39 34

**Comment by amdrew** [comment]
@jasmussen I've updated the PR and removed the alignment controls from the block inspector. 

Good spotting with the responsiveness, I hadn't considered that. Is placing other inspector controls inside the transform menu a pattern that Gutenberg utilizes with other blocks at mobile device widths? 

Though it seems you can scroll horizontally to reach other inspector controls (only tested on iOS simulator for mac), I'm unaware on how other mobile devices treat this - Do they all provide some mechanism to scroll horizontally to reach the other controls? If not, moving the headings into the transform menu does seem like the best compromise.

In any case, I've left the headings alone for now as I'm not sure how to move them inside the transform menu, and only for mobile devices (is there a pre-existing function for this?). If someone can give me some pointers I'll definitely give it a shot.

**Comment by jasmussen** [comment]
> Good spotting with the responsiveness, I hadn't considered that. Is placing other inspector controls inside the transform menu a pattern that Gutenberg utilizes with other blocks at mobile device widths?

In general nothing should be put inside the inspector that is necessary for the basic operation of the block, either on desktop or mobile. In this case, the PR addresses the issue that _alignments_ could be considered necessary for the basic operation of the heading block. I don't disagree with that, and there is consensus that this is the case. 

The horizontal scrollbar on mobile is a "last resort" at making sure all controls are available, but it's not a good pattern and it's not scalable. The responsive desktop breakpoint is more of an issue, because you can't scroll there.

There are ideas of putting alignments inside a dropdown, but that's also a bandaid.

The best solution is to be careful what we put in the block toolbar, and make sure it scales well. In this case, I thin the solution is, as previously noted, to remove the headings from the block toolbar and add them as transformations.

Thank you for your work so far, really appreciate it. @gziolo if you have bandwidth, do you think you can look at putting the heading sizes inside the switcher? And @karmatosed do you agree that's the best approach for now?

**Comment by amdrew** [comment]
Got it, I agree. It's difficult to scroll horizontally on a mobile device unless you know you actually can in the first place.

**Comment by jasmussen** [comment]
So, here are mockups because they speak louder than words. It's not working quite as well visually, as I'd hoped, but it can serve as a basis for discussion.

heading block toolbar

transformations

I imagine you'd still want to have just a single heading block that you inserted, which would default to H2.

**Comment by gziolo** [comment]
The same applies to other blocks:

screen shot 2018-04-04 at 13 32 57

screen shot 2018-04-04 at 13 33 28

**Comment by karmatosed** [comment]
@jasmussen I like these mocks. I do wonder if that's a lot of text in a line and the 'size' is needed, maybe it is. I do like it better than what we have now and think it's worth creating to experience.

**Pull request closed without merging**